typing/typecore.ml

```diff
@@ -1736,8 +1736,7 @@ and is_nonexpansive_opt = function
   | Some e -> is_nonexpansive e
 
 let check_recursive_bindings env valbinds =
-  let ids = List.concat
-      (List.map (fun b -> pat_bound_idents b.vb_pat) valbinds) in
+  let ids = rev_let_bound_idents valbinds in
```

Comment:
why the `rev_` here? Is the order preserved compared to the previous implementation?

(I think that the order might matter for this code, it is probably reasonable to fail on the first incorrect declaration rather than the last one.)

Reply:
I assumed the order didn't matter, that's why I picked the `rev` one.
But yes, let's not force ourselves to think about it: I'll remove the `rev_` prefix and then merge.